core/domain/stats_domain.py

```diff
@@ -686,22 +708,47 @@ def to_dict(self):
         }
 
     @classmethod
-    def from_dict(cls, issue_dict):
-        """Returns an ExplorationIssue object from a dict.
+    def validate_input_data(cls, exp_issue_dict):
+        """Validates incoming ExplorationIssue data
 
         Args:
-            issue_dict: dict. A dict mapping of all fields of ExplorationIssue
-                object.
+            exp_issue_dict: dict. A dict mapping of all fields of
+                ExplorationIssue object.
+
+        Raises:
+            ValidationError in case one of required properties is missed.
+        """
+        exp_issue_properties = [
+            'issue_type', 'schema_version', 'issue_customization_args',
+            'playthrough_ids', 'is_valid']
+        for exp_issue_property in exp_issue_properties:
+            if exp_issue_property not in exp_issue_dict:
+                raise utils.ValidationError(
+                    '%s not in exploration issue dict.' % (
+                        exp_issue_property))
+
+    @classmethod
+    def from_dict(cls, exp_issue_dict):
+        """Checks whether the exploration issue dict has the correct keys and
+        then returns a domain object instance.
+
+        Args:
+            exp_issue_dict: dict. A dict mapping of all fields of
+                ExplorationIssue object.
 
         Returns:
             ExplorationIssue. The corresponding ExplorationIssue domain object.
         """
-        return cls(
-            issue_dict['issue_type'],
-            issue_dict['issue_customization_args'],
-            issue_dict['playthrough_ids'],
-            issue_dict['schema_version'],
-            issue_dict['is_valid'])
+        cls.validate_input_data(exp_issue_dict)
+        exp_issue = cls(
+            exp_issue_dict['issue_type'],
+            exp_issue_dict['issue_customization_args'],
+            exp_issue_dict['playthrough_ids'],
+            exp_issue_dict['schema_version'],
+            exp_issue_dict['is_valid'])
+
+        exp_issue.validate()
+        return exp_issue
 
     @classmethod
     def from_backend_dict(cls, exp_issue_dict):
```

Comment:
Isn't the aim of the PR  to remove this function?

Reply:
Yes, it is. But I decided to keep it for now and raise a warning to anyone who is trying to use it. I feel that it is a safer way to do such refactoring.

Reply:
But, why? Since, this function is not used anywhere and where a similar functionality is needed, you have used `from_dict`. So, wouldn't that notify any contributors to use just `from_dict`?

Reply:
It is not used in my fork, but someone may use it right now in a different fork/branch. Even more, it may be used in some unmerged "green" PRs. As a result, if we merge my PR and some "green" PR which is using the deprecated method, we may have an issue.
Does it make sense?

Reply:
Hi @vpishuk -- I really appreciate you thinking about this! That's excellent engineering.

That said -- let's not do this. Instead, I would rather remove the function directly (and ditto for the other one too). We do have a guard against this in the form of the backend coverage checks which should be at 100%. When you merge your PR, this should cause other PRs to fail the prereq checks, so I think we're safe.

Reply:
Ok. I didn't know that you have this configured.